src/View/Shortcodes/EmbedShortcodeProvider.php

```diff
@@ -89,7 +89,24 @@ public static function handle_shortcode($arguments, $content, $parser, $shortcod
         }
 
         // Process embed
-        $embed = $embed->getEmbed();
+        try {
+            $embed = $embed->getEmbed();
+        }
+        catch(\Embed\Exceptions\InvalidUrlException $e) {
+            $attr = array(
+                'class' => 'ss-media-exception embed'
+            );
+            if (Director::isDev()) {
+                $result = HTML::createTag('div', $attr, $e->getMessage());
+                return $result;
+            }
+            $result = HTML::createTag(
+                'div',
+                $attr,
+                HTML::createTag('p', array(), 'There was a problem loading the media.')
```

Comment:
If changing to the [] format from array() in @ScopeyNZ s suggestion, then probably should do the same here. 

```suggestion
                HTML::createTag('p', [], 'There was a problem loading the media.')
```

Reply:
Should the text be translateable?  
https://docs.silverstripe.org/en/4/developer_guides/i18n/
```suggestion
                HTML::createTag('p', [], _t("Namespace.Entity", "There was a problem loading the media");           
```
@ScopeyNZ  - can you advise what the namespace & entity should be?

Reply:
Good catch. Normally the namespace is related to the class (and namespace) - the "entity" indicates the content of the message. Maybe something like `EmbedShortcodeProvider.INVALID_URL`?

Reply:
EmbedShortcodeProvider.INVALID_URL works for me : )

Reply:
It should be `_t(__CLASS__ . '.INVALID_URL', 'There was a problem loading the media.')` so we get the fully qualified class name, that’s the convention across core code 😄